gems/aws-sigv4/lib/aws-sigv4/signer.rb

```diff
@@ -375,6 +394,19 @@ def string_to_sign(datetime, canonical_request)
         ].join("\n")
       end
 
+      def event_string_to_sign(datetime, headers, payload, prior_signature, encoder)
+        encoded_headers = encoder.encode_headers(
+          Aws::EventStream::Message.new(headers: headers, payload: payload)).read
```

Comment:
indentation here is a bit hard to follow, I'd at least most the last `).read` down to its own line.